modules/lang-painless/src/main/java/org/elasticsearch/painless/node/SFunction.java

```diff
@@ -121,7 +118,8 @@ void generateSignature(PainlessLookup painlessLookup) {
                 PainlessLookupUtility.typeToJavaType(returnType), paramClasses).toMethodDescriptorString());
     }
 
-    void analyze(ScriptRoot scriptRoot) {
+    // TODO: decouple from ir
```

Comment:
Punt to issue.

Reply:
Removed as this TODO no longer makes sense.